api/disruption_types_test.go

```diff
@@ -0,0 +1,222 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package api_test
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("DisruptionStatus.RemoveDeadTargets Test", func() {
+	var matchingTargets []string
+	var status *v1beta1.DisruptionStatus
+
+	BeforeEach(func() {
+		rand.Seed(time.Now().UnixNano())
+		status = &v1beta1.DisruptionStatus{
+			Targets: []string{"0.0.0.0", "1.1.1.1", "2.2.2.2", "3.3.3.3", "4.4.4.4"},
+		}
+	})
+
+	JustBeforeEach(func() {
+		rand.Shuffle(len(matchingTargets), func(i, j int) {
+			matchingTargets[i], matchingTargets[j] = matchingTargets[j], matchingTargets[i]
+		})
+	})
+
+	When("matchingTargets contains some current targets", func() {
+		BeforeEach(func() {
+			matchingTargets = status.Targets
+		})
+
+		When("matchingTargets contains exactly all the current targets", func() {
+			It("expects status.Targets to be intact", func() {
+				status.RemoveDeadTargets(matchingTargets)
+				Expect(status.Targets).To(HaveLen(5))
+				Expect(status.Targets).To(Equal(matchingTargets))
+			})
+		})
+
+		When("matchingTargets has all current targets and more", func() {
+			BeforeEach(func() {
```

Comment:
```suggestion
			JustBeforeEach(func() {
```

to make the timing more explicit

Reply:
done